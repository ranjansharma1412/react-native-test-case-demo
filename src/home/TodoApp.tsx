import React, { useState } from 'react'
import { Button, FlatList, StyleSheet, Text, TextInput, View } from 'react-native'

const TodoApp = () => {
    const [totoName, setTotoName] = useState<string>('');
    const [totoList, setTotoList] = useState<string[]>([]);

    const addTodo = () => {
        setTotoList([...totoList, totoName]);
    }

    const deleteTodo = (item: string) => {
        const itemAfterDeletion = totoList.filter((todoItem) => todoItem !== item);
        setTotoList([...itemAfterDeletion]);
    }

    const _renderTodoItem = ({ item, index }: { item: string, index: number }) => {
        return (
            <View style={styles.todoItemStyle} key={`${index}`}>
                <Text>{item}</Text>
                <Text style={{color:'red'}} onPress={() => deleteTodo(item)}>Delete</Text>
            </View>
        )
    }

    return (
        <View style={styles.root}>
            <Text style={styles.title}>TodoApp</Text>
            <Text style={styles.todoCount}>{`${totoList.length} todo remaining`}</Text>
            <View style={{ marginTop: 20 }}>
                <TextInput
                    style={styles.inputStyle}
                    placeholder='Enter to name'
                    value={totoName}
                    onChangeText={setTotoName}
                />
                <View style={{ marginTop: 20 }}>
                    <Button testID='add-todo' onPress={addTodo} title='Add Todo' />
                </View>
            </View>
            <View style={{ marginTop: 15 }}>
                <FlatList
                    data={totoList}
                    renderItem={_renderTodoItem}
                    keyExtractor={((item, index) => index.toString())}
                />
            </View>
        </View>
    )
}

export default TodoApp

const styles = StyleSheet.create({
    root: {
        flex: 1,
        paddingHorizontal: 20,
    },
    title: {
        fontSize: 20,
        fontWeight: 'bold',
        textAlign: 'center'
    },
    inputStyle: {
        height: 45,
        width: '100%',
        backgroundColor: 'white'
    },
    todoItemStyle: {
        paddingVertical: 5,
        backgroundColor: 'white',
        marginTop: 10,
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'space-between'
    },
    todoCount: {
        fontSize: 16,
        textAlign: 'right',
    }
})