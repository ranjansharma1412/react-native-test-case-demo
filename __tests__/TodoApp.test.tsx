import { it, describe, expect } from '@jest/globals';
// import renderer from 'react-test-renderer';
import { render, fireEvent, screen } from '@testing-library/react-native';
import TodoApp from '../src/home/TodoApp';

describe('Todo app test case',()=>{
    it('todo app snapshot testing',()=>{
        render(<TodoApp/>)
        expect(screen.toJSON()).toMatchSnapshot()
    })
})


