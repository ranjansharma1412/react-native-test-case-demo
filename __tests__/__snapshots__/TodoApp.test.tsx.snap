// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo app test case todo app snapshot testing 1`] = `
<View
  style={
    {
      "flex": 1,
      "paddingHorizontal": 20,
    }
  }
>
  <Text
    style={
      {
        "fontSize": 20,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    TodoApp
  </Text>
  <Text
    style={
      {
        "fontSize": 16,
        "textAlign": "right",
      }
    }
  >
    0 todo remaining
  </Text>
  <View
    style={
      {
        "marginTop": 20,
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Enter to name"
      style={
        {
          "backgroundColor": "white",
          "height": 45,
          "width": "100%",
        }
      }
      value=""
    />
    <View
      style={
        {
          "marginTop": 20,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
        testID="add-todo"
      >
        <View
          style={
            [
              {},
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Add Todo
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "marginTop": 15,
      }
    }
  >
    <RCTScrollView
      data={[]}
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View />
    </RCTScrollView>
  </View>
</View>
`;
